### PR TITLE
Multiple Fixes and Residual Analysis Added

### PR DESCRIPTION
**Pull request by ngupta23**
## Related Issuse or bug
  - #1561
  - #1646
  - #1644
  - #1589
  
Fixes: #[issue number that will be closed through this PR]
  - #1561
  - #1646
  - #1644

#### Describe the changes you've made
- [Enhancement] Adds residual analysis
- [Bug Fix] Metrics are not displayed when model is loaded back after saving.
- [Bug Fix] Defaults for functional API plot model were fixed.
- [Bug Fix]Blending method passed from functional API to OOP call internally now (missing earlier so defaulted to `mean` all the time)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by Yard1 — APPROVED** [review]
Thanks!

**Comment by Yard1** [comment]
@ngupta23 if you are going to go with `hasattr`, there are several other spots in the PR that are using try except as well

**Comment by ngupta23** [comment]
> 
> 
> @ngupta23 if you are going to go with `hasattr`, there are several other spots in the PR that are using try except as well

Yes, I changed it in all spots now except

```python
            try:
                model_name = self._get_model_name(estimator)
            except AttributeError:
                # If the model is saved and loaded afterwards,
                # it will not have self._get_model_name
                model_name = estimator.__class__.__name__
````

I think this exception handling should be inside `self._get_model_name(estimator)` itself. Not sure why time_series module has to do this. What do you think?

File: pycaret/internal/pycaret_experiment/time_series_experiment.py [hunk 1]
```diff
@@ -2736,15 +2752,21 @@ def predict_model(
...
-        X_test_ = self.X_test.copy()
-        # Some predict methods in sktime expect None (not an empty dataframe as
-        # returned by pycaret). Hence converting to None.
-        if X_test_.shape[0] == 0 or X_test_.shape[1] == 0:
-            X_test_ = None
-        y_test_ = self.y_test.copy()
+        display_test_metric = True
+        try:
+            self.X_test
+        except AttributeError:
```
[the remark this answers is not in the document]
Comment (ngupta23) [reply pycaret/internal/pycaret_experiment/time_series_experiment.py]: Fixed in all places except the one noted in the comments.

[the remark this answers is not in the document]
Comment (ngupta23) [reply pycaret/internal/pycaret_experiment/time_series_experiment.py]: > I think this exception handling should be inside `self._get_model_name(estimator)` itself. Not sure why time_series module has to do this. What do you think?

Never mind. I have changed it for now so we can close this out....

**Comment by Yard1** [comment]
@ngupta23 That's a good idea, feel free to put it there!

**Comment by ngupta23** [comment]
> 
> 
> @ngupta23 That's a good idea, feel free to put it there!

I don't think I can. Its a method call, so the calling method will have to check for it - correct? Anyway, I have fixed it in my method now.

**Comment by Yard1** [comment]
@ngupta23 Right, I'll take a look at that.

**Pull request merged**